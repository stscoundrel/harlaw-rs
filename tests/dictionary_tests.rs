use harlaw::{get_dictionary, get_dictionary_without_markup, get_dictionary_with_custom_settings};
use harlaw::{HarlawSettings, ContentReplace};
use insta::assert_json_snapshot;

#[test]
fn gets_default_dictionary() {
    let input = "./tests/fixtures/default_dictionary.dsl";
    let result = get_dictionary(input).unwrap();

    assert_json_snapshot!(result)
}

#[test]
fn gets_dictionary_without_markup() {
    let input = "./tests/fixtures/default_dictionary.dsl";
    let result = get_dictionary_without_markup(input).unwrap();

    assert_json_snapshot!(result)
}

#[test]
fn gets_dictionary_with_custom_settings() {
    let input = "./tests/fixtures/default_dictionary.dsl";
    let settings = HarlawSettings {
        removes: vec!["[m1]", "[m2]", "[/m]", "\t"],
        replaces: vec![
            ContentReplace {
                search: "[b]",
                replace: "<TUHTI>",
            },
            ContentReplace {
                search: "[/b]",
                replace: "</TUHTI>",
            },
            ContentReplace {
                search: "[i]",
                replace: "<VINO>",
            },
            ContentReplace {
                search: "[/i]",
                replace: "</VINO>",
            }
        ],
    };
    let result = get_dictionary_with_custom_settings(input, settings).unwrap();

    assert_json_snapshot!(result)
}