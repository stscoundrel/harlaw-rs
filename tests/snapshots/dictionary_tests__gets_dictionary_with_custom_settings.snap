---
source: tests/dictionary_tests.rs
expression: result

---
[
  {
    "word": "foo",
    "definitions": [
      "Lorem ipsum dolor sit amet, <VINO>dolor</VINO> sit igitur"
    ]
  },
  {
    "word": "bar",
    "definitions": [
      "<TUHTI>Dolor</TUHTI> sit igitur.",
      "Lorem ipsum dolor sit amet."
    ]
  },
  {
    "word": "baz",
    "definitions": [
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
    ]
  },
  {
    "word": "foo2",
    "definitions": [
      "Lorem ipsum dolor sit amet, <VINO>dolor</VINO> sit igitur"
    ]
  },
  {
    "word": "bar2",
    "definitions": [
      "<TUHTI>Dolor</TUHTI> sit igitur.",
      "Lorem ipsum dolor sit amet."
    ]
  },
  {
    "word": "baz2",
    "definitions": [
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
    ]
  }
]
